### PR TITLE
Adding 2016 pre-VFP entries (v12)

### PR DESCRIPTION
**Pull request by aalvesan**
Adding the campaign run2_2016_HIPM_uhh_v12. Includes all cmsdb entries for data, signal and background for 2016 nanoAOD datasets, at version 12, and for pre-VFP era.

### CODE REVIEW

File: cmsdb/campaigns/run2_2016_HIPM_uhh_v12/hh2bbtautau.py [hunk 1]
```diff
@@ -0,0 +1,1274 @@
...
+        "/VBFToBulkGravitonToHHTo2B2Tau_M-750_TuneCP5_PSWeights_narrow_13TeV-madgraph-pythia8/RunIISummer20UL16MiniAODAPVv2-106X_mcRun2_asymptotic_preVFP_v11-v2/NANOAODSIM",  # noqa
+    ],
+    n_files=1,
+    n_events=106000,
+)
+
+cpn.add_dataset(
+    name="graviton_hh_vbf_bbtautau_m700_madgraph",
+    id=14309734,
+    processes=[procs.graviton_hh_vbf_bbtautau_m700_madgraph],
```
Comment (pkausw) [inline cmsdb/campaigns/run2_2016_HIPM_uhh_v12/hh2bbtautau.py:1048]:
```suggestion
    processes=[procs.graviton_hh_vbf_bbtautau_m700],
```

File: cmsdb/campaigns/run2_2016_HIPM_uhh_v12/hh2bbtautau.py [hunk 2]
```diff
@@ -0,0 +1,1274 @@
...
+        "/VBFToBulkGravitonToHHTo2B2Tau_M-1750_TuneCP5_PSWeights_narrow_13TeV-madgraph-pythia8/RunIISummer20UL16MiniAODAPVv2-106X_mcRun2_asymptotic_preVFP_v11-v2/NANOAODSIM",  # noqa
+    ],
+    n_files=1,
+    n_events=54000,
+)
+
+cpn.add_dataset(
+    name="graviton_hh_vbf_bbtautau_m1500_madgraph",
+    id=14309566,
+    processes=[procs.graviton_hh_vbf_bbtautau_m1550],
```
Comment (pkausw) [inline cmsdb/campaigns/run2_2016_HIPM_uhh_v12/hh2bbtautau.py:1246]:
```suggestion
    processes=[procs.graviton_hh_vbf_bbtautau_m1500],
```

**Review by pkausw — CHANGES_REQUESTED** [review]
Hi @aalvesan , thanks for this massive amount of work! All in all, it looks in good shape :+1: I have some small requests:
- there are typos here and there in the different files. I did my best to check everything carefully, but can you also have another look?
- I think the imports in the different files point to the 2017 campaign, please correct this
- Therse seems to be an issue with datasets that or present twice. Might be connected to extenstions of datasets, which logically should belong to the dataset w/o the `ext` tag
- Can you also solve the linting errors on your branch?

I'm also tagging @dsavoiu here, because I know he also put some thought and work into the naming scheme of processes and datasets. Maybe he'll have additional comments

File: cmsdb/campaigns/run2_2016_HIPM_uhh_v12/Signals.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
+# coding: utf-8
```
[the remark this answers is not in the document]
Comment (aalvesan) [reply cmsdb/campaigns/run2_2016_HIPM_uhh_v12/Signals.py:1]: this was the wrong file, sorry.

File: cmsdb/campaigns/run2_2016_HIPM_uhh_v12/MultiBoson.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+cpn.add_dataset(
+    name="zzz_ext_amcatnlo",
+    id=14215019,
+    processes=[procs.zzz_ext],
+    keys=[
+        "/ZZZ_TuneCP5_13TeV-amcatnlo-pythia8/RunIISummer20UL16MiniAODAPVv2-106X_mcRun2_asymptotic_preVFP_v11_ext1-v1/NANOAODSIM",  # noqa
+    ],
+    n_files=5,
+    n_events=5302000,
+)
```
[the remark this answers is not in the document]
Comment (aalvesan) [reply cmsdb/campaigns/run2_2016_HIPM_uhh_v12/MultiBoson.py:31]: one should be zzz and the other zzz_ext, but I did zzz_ext twice.... I will fix it

File: cmsdb/campaigns/run2_2016_HIPM_uhh_v12/MultiBoson.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+cpn.add_dataset(
+    name="www_ext_amcatnlo",
+    id=14215136,
+    processes=[procs.www_ext],
+    keys=[
+        "/WWW_4F_TuneCP5_13TeV-amcatnlo-pythia8/RunIISummer20UL16MiniAODAPVv2-106X_mcRun2_asymptotic_preVFP_v11_ext1-v1/NANOAODSIM",  # noqa
+    ],
+    n_files=5,
+    n_events=5190000,
+)
```
[the remark this answers is not in the document]
Comment (aalvesan) [reply cmsdb/campaigns/run2_2016_HIPM_uhh_v12/MultiBoson.py:86]: same mistake here

File: cmsdb/campaigns/run2_2016_HIPM_uhh_v12/__init__.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    bx=25,
+    aux={
+        "tier": "NanoAOD",
+        "year": 2016,
+        "identifier": "HIMP",
+        "version": 12,
+        "custom": {
+            "name": "run2_2016_HIMP_nano_uhh_v12",
+            "creator": "uhh",
+            "location": "davs://dcache-cms-webdav-wan.desy.de:2880/pnfs/desy.de/cms/tier2/store/user/bwieders/nano_uhh_v11/merged_2048.0MB",  # noqa
```
[the remark this answers is not in the document]
Comment (aalvesan) [reply cmsdb/campaigns/run2_2016_HIPM_uhh_v12/__init__.py:37]: yes, I fixed it

**Review by dsavoiu — CHANGES_REQUESTED** [review]
Hi @aalvesan, thanks for adding this campaign! And thanks @pkausw for the ping ;)

I only noticed some minor things (naming conventions, typos), but maybe you could have a look and make changes where necessary?
 
Also, in general, I think it would be good to only use lowercase or "snake-case" for filenames in `cmsdb`, so instead of `MultiBoson.py`, could you use `multi_boson.py` or similar? Same for one-letter filenames (`Z.py` -> `z.py`, etc.)

Actually, I think in general in the other campaigns we collected the Drell-Yan, Multiboson, Z and W background samples in one electroweak file `ewk.py`. Unless we have a good reason for the separation, I would propose to rename or merge following files into one:
* `DY.py`, `MultiBoson.py`, `W.py`, `Z.py` -> `ewk.py`
* `SingleHiggs.py` -> `higgs.py`
* `SingleTop.py` -> `st.py`

File: cmsdb/campaigns/run2_2016_HIPM_uhh_v12/MultiBoson.py [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+        "/ZZZ_TuneCP5_13TeV-amcatnlo-pythia8/RunIISummer20UL16MiniAODAPVv2-106X_mcRun2_asymptotic_preVFP_v11-v1/NANOAODSIM",  # noqa
+    ],
+    n_files=1,
+    n_events=81000,
+)
+
+cpn.add_dataset(
+    name="zzz_ext_amcatnlo",
+    id=14215019,
+    processes=[procs.zzz_ext],
```
[the remark this answers is not in the document]
Comment (aalvesan) [reply cmsdb/campaigns/run2_2016_HIPM_uhh_v12/MultiBoson.py:25]: okay, I will include all of your suggestions. Thanks for the review! :)

**Review by dsavoiu — CHANGES_REQUESTED** [review]
Hi @aalvesan, thanks for implementing the changes! Everything looks fine now, I just noticed a few minor things that should be easy to fix (see below).

**Comment by dsavoiu** [comment]
Hi @aalvesan, sorry for the delay in reviewing this. I actually noticed that I had overlooked some things in the initial review and also that the tests were failing due to some typos in the imports and missing processes. I took the liberty of committing a series of changes to fix these issues, so this should in principle now be ready to merge.

But @aalvesan please check if the changes work for you and I would also ping @riga and @pkausw here in case they also want to look over the changes before the merge.

**Comment by dsavoiu** [comment]
Hi @aalvesan, I fixed the conflicts with the `master` branch, which were due to the restructuring of unit tests introduced by #22. Did you manage to have a look at the latest changes? If there are no further things to do we could go ahead and merge this soon.

**Review by riga — APPROVED** [review]
LGTM!

**Review by dsavoiu — APPROVED** [review]

**Pull request merged**